cmd/collector/app/sampling/strategystore/factory.go

```diff
@@ -27,8 +30,11 @@ import (
 // plugin.Configurable
 type Factory interface {
 	// Initialize performs internal initialization of the factory.
-	Initialize(metricsFactory metrics.Factory, logger *zap.Logger) error
+	Initialize(metricsFactory metrics.Factory, lock distributedlock.Lock, store samplingstore.Store, logger *zap.Logger) error
 
 	// CreateStrategyStore initializes the StrategyStore and returns it.
-	CreateStrategyStore() (StrategyStore, error)
+	CreateStrategyStore() (StrategyStore, Aggregator, error)
+
+	// RequiresLockAndSamplingStore indicates whether this strategy store requires a Lock and SamplingStore
+	RequiresLockAndSamplingStore() (bool, error)
```

Comment:
I'm not clear on the purpose of this. First, how can a factory for sampling store not require a sampling store? Second, who calls this and what changes depending on the output? 

Let's add better docs.